### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by Akeron44**
I added the sign up and login logic.

### CODE REVIEW

**Review by danielaKokoneshi309 — APPROVED** [review]

**Review by gerimujo496 — APPROVED** [review]

File: models/user.js [hunk 1]
```diff
@@ -1,11 +1,44 @@
...
+  return token;
+};
+
 const User = mongoose.model("User", userSchema);
+
+function validateUser(user) {
+  const schema = Joi.object({
+    firstName: Joi.string().min(3).required(),
+    lastName: Joi.string().min(3).required(),
+    email: Joi.string().min(5).required().email(),
```
Comment (KostandinD) [inline models/user.js]: Hey, let's keep up the consistency and do a minor change in the chaining order from `Joi.string().min(5).required().email()` to `Joi.string().min(5).email().required()`.

File: models/user.js [hunk 2]
```diff
@@ -1,11 +1,44 @@
...
+
+    process.env.JWT_PRIVATE_KEY,
+    { expiresIn: "10h" }
+  );
+  return token;
+};
+
 const User = mongoose.model("User", userSchema);
+
+function validateUser(user) {
```
Comment (KostandinD) [inline models/user.js]: I recommend modularising the code by placing it in a separate file, such as **validation** or **userSchemaValidation**, based on the upcoming code.

File: routes/signup.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  let user = await User.findOne({ email: req.body.email });
+
+  if (user) {
+    return res.status(400).send({ message: "User already registered" });
+  }
+
+  const newUser = new User(
+    _.pick(req.body, ["firstName", "lastName", "email", "password"])
+  );
+  const salt = await bcrypt.genSalt(10);
```
Comment (KostandinD) [inline routes/signup.js:23]: Please add spacing when it makes sense.

File: startup/routes.js [hunk 4]
```diff
@@ -1,13 +1,11 @@
-
 const express = require("express");
-
+const signup = require("../routes/signup");
+const login = require("../routes/login");
 
 require("express-async-errors");
```
Comment (KostandinD) [inline startup/routes.js]: It would be better if you would place this import on top of the file.

**Review by KostandinD — CHANGES_REQUESTED** [review]
The work done so far seems nice, however we must oblige by clean code rules and standards.

File: models/user.js (hunk 2 above)
Comment (Akeron44) [reply models/user.js]: Thank you, I created a second commit based on your review.

**Comment by KostandinD** [comment]
Thank you for the prompt reaction. Code looks much better now 😄. 
An advice for the upcoming commits, please use meaningful commit messages. 
For instance, in the case of the second commit the message could have been: "Code refactoring". This message is more descriptive, therefore it provides for me a better understanding of what is going on in the commit.

**Pull request merged**